File: d587(delicious)/Python/delicious.py

while True:
    try:
        num_of_nums = int(input())

        nums = list(map(int, input().split()))

        buckets = [0, 0, 0]

        for n in nums:
            buckets[n % 3] += 1

        for i in range(buckets[1]):
            print(1, end = ' ')
        for i in range(buckets[2]):
            print(2, end = ' ')
        for i in range(buckets[0]):
            print(3, end = ' ')
        print()
    except:
        break

File: d587(delicious)/C++/delicious.cpp
#include <iostream>

int main(void) {
  int num_of_nums;
  while (std::cin >> num_of_nums) {
    int buckets[3] = {0};
    for(int i = 0; i < num_of_nums; i++) {
      int num;
      std::cin >> num;
      buckets[num % 3]++;

    }
    for(int i = 0; i < buckets[1]; i++) {
      std::cout << "1 ";
    }
    for(int i = 0; i < buckets[2]; i++) {
      std::cout << "2 ";
    }
    for(int i = 0; i < buckets[0]; i++) {
      std::cout << "3 ";
    }
    std::cout << '\n';
  }

  return 0;
}
